runtime/win32.c

```diff
@@ -622,20 +624,33 @@ void caml_win32_unregister_overflow_detection(void)
 
 int caml_win32_random_seed (intnat data[16])
 {
-  /* For better randomness, consider:
-     http://msdn.microsoft.com/library/en-us/seccrypto/security/rtlgenrandom.asp
-     http://blogs.msdn.com/b/michael_howard/archive/2005/01/14/353379.aspx
-  */
+  int n = 0;
+  unsigned char buffer[12];
+  int nread = 0;
+
+  /* Try kernel entropy first */
+  NTSTATUS st = BCryptGenRandom(NULL, buffer, 12,
+                                BCRYPT_USE_SYSTEM_PREFERRED_RNG);
+  if (NT_SUCCESS(st))
+    nread = 12;
+  while (nread > 0) data[n++] = buffer[--nread];
```

Comment:
Why not `memcpy(data, buffer, sizeof(buffer))`?

Also, this loop can be put under the `if` above, and be made to follow by the "return" below (always unde the `if`).

Reply:
> Why not `memcpy(data, buffer, sizeof(buffer))`?

The function doesn't return the number of random bytes written to `data`, it returns the number of `intnats` that have random `unsigned char` in their low-byte. If I do a memcpy and then return `12`, the final OCaml seed array will have 12 intnats, of which the first 12 bytes will be random data, and the rest (I guess) the stack values that was there (or zeroes?).